File: 1-100/75-sort-colors.cpp
void sortColors(vector<int>& nums) {
    int zero = 0, two = nums.size() - 1;
    int curr = 0;
    while (curr <= two) {
        if (nums[curr] == 0) {
            swap(nums[curr], nums[zero]);
            ++curr;
            ++zero;
        }
        else if (nums[curr] == 1)
            ++curr;
        else if (nums[curr] == 2) {
            swap(nums[curr], nums[two]);
            --two;
        }
    }
}

File: python version/75-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero = 0
        two = len(nums)-1
        curr = 0
        while curr <= two:
            if nums[curr] == 0:
                nums[curr], nums[zero] = nums[zero], nums[curr]
                curr += 1
                zero += 1
            elif nums[curr] == 2:
                nums[curr], nums[two] = nums[two], nums[curr]
                two -= 1
            else:
                curr += 1
        